<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <group ns="dodobot" >
        <node name="db_detectnet" pkg="db_detectnet" type="db_detectnet_node" output="log">
            <param name="model_name" value=""/>

            <param name="model_path" value="$(find db_detectnet)/models/dodobot_objects_mb2-ssd-lite/mb2-ssd-lite.onnx"/>
            <param name="class_labels_path" value="$(find db_detectnet)/models/dodobot_objects_mb2-ssd-lite/labels.txt"/>

            <!-- <param name="model_path" value="$(find db_detectnet)/models/dodobot_objects_ssd_resnet50_v1_fpn/model.onnx"/> -->
            <!-- <param name="class_labels_path" value="$(find db_detectnet)/models/dodobot_objects_ssd_resnet50_v1_fpn/labels.txt"/> -->

            <param name="prototxt_path" value=""/>
            <param name="input_blob" value="input_0"/>
            <param name="output_cvg" value="scores"/>
            <param name="output_bbox" value="boxes"/>
            <param name="overlay_flags" value="box,labels,conf"/>
            <param name="mean_pixel_value" value="0.0"/>
            <param name="threshold" value="0.3"/>

            <param name="min_valid_dist" value="0.06"/>
            <param name="max_valid_dist" value="0.75"/>

            <param name="depth_topic"           value="/camera/aligned_depth_to_color/image_raw"/>
            <!-- <param name="depth_topic"           value="/camera/depth_decimate/image_raw"/> -->
            <!-- <param name="depth_topic"           value="/camera/depth/image_rect_raw"/> -->

            <param name="depth_info_topic"      value="/camera/aligned_depth_to_color/camera_info"/>
            <!-- <param name="depth_info_topic"           value="/camera/depth_decimate/camera_info"/> -->
            <!-- <param name="depth_info_topic"      value="/camera/depth/camera_info"/> -->

            <param name="color_topic"     value="/camera/color/image_raw"/>
            <!-- <param name="color_topic"     value="/camera/color_decimate/image_raw"/> -->
            <param name="color_info_topic"     value="/camera/color/camera_info"/>
            <!-- <param name="color_info_topic"     value="/camera/color_decimate/camera_info"/> -->

            <param name="approx_sync"           value="false"/>
            <param name="publish_debug_image"     value="true"/>

            <!-- <rosparam param="detectnet_labels" file="$(find db_detectnet)/config/labels.yaml" command="load"/> -->
            <rosparam param="detectnet_marker_colors" file="$(find db_detectnet)/config/marker_colors.yaml" command="load"/>
            <rosparam param="detectnet_z_depth_estimations" file="$(find db_detectnet)/config/depth_estimations.yaml" command="load"/>

        </node>
    </group>
</launch>
